### ISSUE TITLE
[KED-1111] Top-level keys in parameters.yml assigned ``null`` value throw DataSetError when referenced as node input

### ISSUE THREAD

**Issue by khdlim** [quantumblacklabs/kedro#117]
## Description
When setting a parameter (e.g. "myparam" in ``parameters.yml`` to ``None``, a ``DataSetError`` is thrown.
``MemoryDataSet`` tries to call ``MemoryDataSet:_load()`` which throws the error as it does not expect ``None`` to be a valid data value.

## Context
I might need to pass ``"params:myparam"`` into a node input and have its value resolve to ``None``. Currently it looks like it may not be possible.

## Steps to Reproduce
1. Create a new project with ``kedro new``.
2. Add line ``myparam: null`` in ``conf/base/parameters.yml``
3. In ``pipeline.py``, do this in ``create_pipelines``:
```
def create_pipelines(**kwargs) -> Dict[str, Pipeline]:
    """Create the project's pipeline.

    Args:
        kwargs: Ignore any additional arguments added in the future.

    Returns:
        A mapping from a pipeline name to a ``Pipeline`` object.

    """

    return {
        "__default__": Pipeline([node(lambda x:x, "params:myparam", "output")])
    }
```
4. Run it via kedro run and error is thrown.

## Expected Result
Ideally, it should load ``None`` and ``"output"`` should be assigned the value ``None``.

## Actual Result
DataSetError is thrown (last few lines of trace reported here)

  File "/home/khdlim/miniconda3/envs/ds/lib/python3.6/site-packages/kedro/io/data_catalog.py", line 326, in load
    return func()
  File "/home/khdlim/miniconda3/envs/ds/lib/python3.6/site-packages/kedro/io/core.py", line 217, in load
    return self._load()
  File "/home/khdlim/miniconda3/envs/ds/lib/python3.6/site-packages/kedro/io/memory_data_set.py", line 83, in _load
    raise DataSetError("Data for MemoryDataSet has not been saved yet.")

## Your Environment
Kedro 15.2, Ubuntu 16.04.6, Python 3.6

**Comment by lorenabalan** [comment]
Hi @khdlim, thank you for raising this, we'll investigate and get back to you. I've tagged this with our internal ticket number to make it easier to track.

**Comment by lorenabalan** [comment]
Hey @khdlim, apologies for getting back to you so late. I just wanted to better understand your use case - is there a reason why you couldn't use special values like `false` or `-1` for example, in your `parameters.yml`, instead of `null`?

As for the saving, Kedro doesn't allow saving `None` to any dataset, not just `MemoryDataSet`, as it's equivalent to an empty dataset. We'd rather the user explicitly do this operation and return empty table/dataset, otherwise non-pure functions that don't return anything might slip through the cracks.

**Comment by lorenabalan** [comment]
Hi @khdlim addressed this in bb0ac32fb37e2799c0140e573f63684ebe0e2bc1 .
In the next release of Kedro you should be able to set `null` in your `parameters.yml`. I've also added a line in the documentation about saving `None`, as you suggested.
Hope this helps! :)
